src/devices/Is31fl3730/Is31fl3730.cs

```diff
@@ -0,0 +1,447 @@
+        public Is31fl3730(I2cDevice i2cDevice, int width, int height)
+        : this(i2cDevice)
+        {
+            Width = width;
+            Height = height;
+        }
+
+        /*
+        The Pimoroni devices support three I2C devices:
+
+            - 0x61 (the default)
+            - 0x62
+            - 0x63
+
+        For the breakout, this is straightforward. There is a default and you can change the address (in the normal way).
+
+        For the Micro Dot pHAT, there are three pairs installed (making six matrices). The addresses are ordered this way:
+
+           x63       x62        x61
+        ________  _________  ________
+        m2 | m1 || m2 | m1 || m2 | m1
+
+        This ordering makes sense if you are scrolling content right to left.
+
+        More detailed information about the structure of each pair follows (further down).
+
+        */
+
+        /// <summary>
+        /// Default I2C address for device.
+        /// </summary>
+        public static readonly int DefaultI2cAddress = 0x61;
+
+        /// <summary>
+        /// Width of LED matrix (x axis).
+        /// </summary>
+        public readonly int Width = 5;
+
+        /// <summary>
+        /// Height of LED matrix (y axis).
+        /// </summary>
+        public readonly int Height = 7;
+
+        /// <summary>
+        /// Brightness of LED matrix (override default value (40 mA); set before calling Initialize method).
+        /// </summary>
+        public int Brightness = 0;
+
+        /// <summary>
+        /// Full current setting for each row output of LED matrix (override default value (128; max brightness); set before calling Initialize method).
+        /// </summary>
+        public int CurrentSetting = 0;
+
+        /// <summary>
+        /// Indexer for matrix.
+        /// </summary>
+        public Matrix5x7? this[int matrix] => _matrices[matrix];
+
+        /// <summary>
+        /// Initialize LED driver.
+        /// </summary>
+        public void Initialize()
+        {
+            if (_configurationValue > 0)
+            {
+                WriteConfiguration();
+            }
+
+            if (CurrentSetting > 0)
+            {
+                _i2cDevice.Write(new byte[] { LIGHTING_EFFECT_REGISTER, (byte)CurrentSetting });
+            }
+
+            if (Brightness > 0)
+            {
+                _i2cDevice.Write(new byte[] { PWM_REGISTER, (byte)Brightness });
+            }
+
+            _matrices[0] = _matrix1Enabled ? new Matrix5x7(this, 0) : null;
+            _matrices[1] = _matrix2Enabled ? new Matrix5x7(this, 1) : null;
+        }
+
+        /// <summary>
+        /// Fill all LEDS.
+        /// </summary>
+        public void Fill(int matrix, int value)
+        {
+            byte[] buffer = _buffers[matrix];
+
+            if (value > 0)
+            {
+                buffer.AsSpan().Fill(255);
+            }
+            else
+            {
+                buffer.AsSpan().Fill(0);
+            }
```

Comment:
```suggestion
            buffer.AsSpan().Fill(value > 0 ? 255 : 0);
```
More a matter of taste, but IL-size is also smaller (which I don't believe is a criteria here).

Or
```suggestion
            if (value > 0)
            {
                buffer.AsSpan().Fill(255);
            }
            else
            {
                buffer.AsSpan().Clear();
            }
```
`Clear` is (in general) better than `Fill(0)` (better optimization in the implementation).

Reply:
just FYI we prefer cleaner code to microoptimizations in this code-base except where it really matters. Allocations - yes, microoptimization not quite. Here I2C transmission itself will likely take more than 99% of the time and this won't make any observable difference. I'd rather write this code as:

```csharp
buffer.AsSpan().Fill(value > 0 ? 255 : 0);
```